fitbenchmarking/results_processing/tests/test_performance_profiler.py

```diff
@@ -33,6 +43,30 @@ def load_mock_results():
             for v in lst]
 
 
+def remove_ids_and_src(html_path):
+    """
+    Remove ids and src within html file.
+    :param html_path: path to html file
+    :type html_path: str
+
+    :return: Processed lines from html file
+    :rtype: list[str]
+    """
+
+    with open(html_path, 'r', encoding='utf-8') as f:
+        read_lines = f.readlines()
+
+    processed_lines = []
+    for str_i in read_lines:
+        pattern_for_ids = r"\b((?:[a-z]+\S*\d+|\d\S*[a-z]+)[a-z\d_-]*)\b\w+"
+        line_without_ids = re.sub(pattern_for_ids, '', str_i)
+        pattern_for_src = r"\b(([A-Za-z._-]+[\\/]+){1,}([A-Za-z.\s_-]+))\b"
```

Comment:
I'm not sure I quite understand the format you're extracting with these. In the first, matching `\S` seems excessive when we know the only symbols will be `-` (as far as I can tell?).

In the second, I think the `{1,}` syntax is the same as `+`? I'm also not sure the source will have any whitespace in the filename?

I'd suggest something like the following is more specific but I don't know what you were testing it against?

```
r'"(?:(?:\d+[a-z]+|[a-z]+\d+)[a-z\d]*-)+[a-z0-9]+"'
and
r'"(?:[A-Za-z_-]+[\\/]+)+[A-Za-z_-]+\.[A-Za-z]+"'
```

Reply:
I tried the patterns you suggested, I had to change the one for the id to r"(?:(?:\d+[a-z]+|[a-z]+\d+)[a-z\d-]*)+[a-z0-9]+",  but like this looks like it works